{{ cookiecutter.formal_name }}/app/src/main/java/org/beeware/android/MainActivity.java

```diff
@@ -92,6 +94,29 @@ private File ensureDir(File dir) throws IOException {
     }
 
     private void unpackPython(Map<String, File> paths) throws IOException {
+        // Try to find `lastUpdateTime` on disk; compare it to actual `lastUpdateTime` from package manager.
+        // https://developer.android.com/reference/android/content/pm/PackageInfo.html#lastUpdateTime
+        Context context = this.getApplicationContext();
+        File lastUpdateTimeFile = new File(context.getCacheDir(), "last-update-time");
+        String storedLastUpdateTime = null;
+        String actualLastUpdateTime = null;
+
+        if (lastUpdateTimeFile.exists()) {
+            BufferedReader reader = new BufferedReader(new InputStreamReader(new
+FileInputStream(lastUpdateTimeFile), StandardCharsets.UTF_8));
+            storedLastUpdateTime = reader.readLine();
+        }
+        try {
+            PackageInfo packageInfo = context.getPackageManager().getPackageInfo(context.getPackageName(), 0);
+            actualLastUpdateTime = String.valueOf(packageInfo.lastUpdateTime);
+        } catch (PackageManager.NameNotFoundException e) {
+            Log.e(TAG, "Unable to find package; using default actualLastUpdateTime: " + actualLastUpdateTime);
```

Comment:
Won't the "default" will always be ``null``? If so, why record the value in the log?

Reply:
No real reason. I just figured I'd print it for, uh, clarity?